saged/models.py

```diff
@@ -19,6 +19,52 @@
 from saged.datasets import LabeledDataset, UnlabeledDataset, MixedDataset, ExpressionDataset
 
 
+def embed_data(unsupervised_config: dict,
+               all_data: MixedDataset,
+               train_data: LabeledDataset,
+               unlabeled_data: UnlabeledDataset,
+               val_data: LabeledDataset,
+               ) -> Tuple[LabeledDataset, LabeledDataset, "UnsupervisedModel"]:
+    """
+    Initialize an unsupervised model and use it to reduce the dimensionality of the data
+    Arguments
+    ---------
+    unsupervised_config: The path to the yml file detailing how to initialize the
+                         unsupervised model
+    all_data: The object storing the data for the entire dataset
+    train_data: The subset of the data to be used for training
+    unlabeled_data: The subset of the data that doesn't have labels
+    val_data: The subset of the data that will be used for validation. To avoid data leakage, the
+              validation data will not be used to train the unsupervised embedding, but will be
+              embedded
+    Returns
+    -------
+    train_data: The embedded training data
+    val_data: The embedded validation data
+    unsupervised_model: The fitted version of the model
+    """
+    # Initialize the unsupervised model
+    unsupervised_model_type = unsupervised_config.pop('name')
+    UnsupervisedClass = globals()[unsupervised_model_type]
```

Comment:
I was initially confused when I saw this, then I remembered that you have the comment later in this file explaining what you're doing here.

Could you write a short syntactic sugar function to do this? Call it `get_class_from_current_file()` or something like that, and just have it return this one-liner, then call it everywhere you need to get a class dynamically? The alternative would be to copy the comment everywhere you use this, but that seems like it might get a bit repetitive.

Reply:
That's a good idea. I thought about copying the comment but chose not to since it was three lines long. Wrapping them in a function makes more sense so I went ahead and did so